docking.js

```diff
@@ -1157,7 +1152,8 @@ const DockedDash = new Lang.Class({
      */
     _adjustPanelCorners: function() {
         let extendHeight = this._settings.get_boolean('extend-height');
-        if (!this._isHorizontal && this._isPrimaryMonitor() && extendHeight && this._fixedIsEnabled) {
+        if (!this._isHorizontal && (this._isPrimaryMonitor() || this._settings.get_boolean('multi-monitor'))
+            && extendHeight && this._fixedIsEnabled) {
```

Comment:
Is this executed on all docks? Another candidate to be moved at the manager level?